anitya/lib/utilities.py

```diff
@@ -77,33 +78,32 @@ def check_project_release(project, session, test=False):
         raise
 
     if test:
-        return up_version
-
-    p_version = project.latest_version or ''
-
-    if up_version:
-        project.logs = 'Version retrieved correctly'
-
-    if up_version and up_version not in project.versions:
-        publish = True
-        project.versions_obj.append(
-            models.ProjectVersion(
-                project_id=project.id,
-                version=up_version
+        return versions
+
+    # There is always at least one version retrieved,
+    # otherwise backedn throws exception
+    project.logs = 'Version retrieved correctly'
+
+    p_versions = project.get_sorted_version_objects()
+    if p_versions:
+        old_version = p_versions[0].version
+
+    for version in versions:
+        if version not in project.versions:
+            project.versions_obj.append(
+                models.ProjectVersion(
+                    project_id=project.id,
+                    version=version
+                )
             )
-        )
 
-    odd_change = False
-    if up_version and up_version != p_version:
-        version_class = project.get_version_class()
-        max_version = max(up_version, p_version, key=version_class)
-        if project.latest_version and max_version != up_version:
-            odd_change = True
-            project.logs = 'Something strange occured, we found that this '\
-                'project has released a version "%s" while we had the latest '\
-                'version at "%s"' % (up_version, project.latest_version)
-        else:
-            project.latest_version = up_version
+    sorted_versions = project.get_sorted_version_objects()
+    max_version = sorted_versions[0].version
+    if project.latest_version != max_version:
+        project.latest_version = max_version
+        publish = True
+    else:
+        project.logs = 'No new version found'
```

Comment:
This depends on the sorting mechanism of the project. If we use RPM version scheme, we will only notify about new version according to RPM version scheme.

Right now we have two version schemes - RPM and Date (Date of retrieval).

But yes, it should be better if the user will get notified about any new version. But the first check must be different because otherwise we will end up with the-new-hotness creating huge amount of bugzilla issues.

The question is how to handle situation, where we receive more than on version at once? Should we notify about every version or only newest?

About the series of projects, it will be good if user could create more than one version series and each of them could have they own version scheme, than it will be easier to identify new versions. It will also be good if the project changed the version scheme on the fly - this sometime happens.

Reply:
This depends on the sorting mechanism of the project. If we use RPM version scheme, we will only notify about new version according to RPM version scheme.

Right now we have two version schemes - RPM and Date (Date of retrieval).

But yes, it should be better if the user will get notified about any new version. But the first check must be different because otherwise we will end up with the-new-hotness creating huge amount of bugzilla issues.

The question is how to handle situation, where we receive more than on version at once? Should we notify about every version or only newest?

About the series of projects, it will be good if user could create more than one version series and each of them could have they own version scheme, than it will be easier to identify new versions. It will also be good if the project changed the version scheme on the fly - this sometime happens.